configs/nosite-x86_64/variables.yaml

```diff
@@ -4,9 +4,10 @@
 # SPDX-License-Identifier: Apache-2.0
 
 variables:
-  batch_time: ''
-  mpi_command: 'mpirun -n {n_nodes} -c {n_ranks} --oversubscribe'
-  batch_submit: '{execute_experiment}'
-  batch_nodes: ''
-  batch_ranks: ''
-  batch_timeout: ''
+  mpi_command: 'mpirun -n {n_ranks} --oversubscribe'
+  n_ranks: '7'
+  n_nodes: '7'
+  n_threads: '7'
+  scheduler: "slurm"
+  batch_submit: "placeholder"
+  mpi_command: "placeholder"
```

Comment:
n_ranks, n_nodes, n_threads, batch_submit, and mpi_command must all be defined (for the first three, they must be positive integers), so I'm thinking each system can define them, and individual experiments can skip defining any that they want the modifier to handle.

Reply:
I think we need n_cores_per_node.  Then n_threads_per_node can default to n_cores_per_node, or people can indicate n_threads_per_node = 2 * n_cores_per_node if they have hyper threads.

n_ranks_per_node can default to n_cores_per_node (or n_gpus_per_node on accelerated systems).

n_ranks = n_nodes * n_ranks_per_node.

Reply:
Sorry here I'm talking about what Ramble needs defined up front.

I agree we should also require that users specify the number of CPUs per node.

Everything under "# Descriptions of resources available on systems" in `class AllocOpt(Enum):` should probably be defined on each system (unless e.g. that system doesn't have GPUs).

Reply:
Thanks, that distinction between "what the machine has" and "what ramble needs to schedule the job" is more clear now.